lib/web/apiserver.go

```diff
@@ -2050,9 +2050,11 @@ func (h *Handler) githubCallback(w http.ResponseWriter, r *http.Request, p httpr
 			logger.Debug("GitHub WebSession created with device web token")
 			// if a device web token is present, we must send the user to the device authorize page
 			// to upgrade the session.
-			// TODO (avatus) the web client currently doesn't handle any redirects after authorizing a web
-			// session with device trust. Once it does, append a redirect_url here as a query parameter
-			return fmt.Sprintf("/web/device/authorize/%s/%s", dwt.Id, dwt.Token)
+			redirectPath, err := BuildDeviceWebRedirectPath(dwt, res.ClientRedirectURL)
+			if err == nil {
+				return redirectPath
+			}
+			logger.WithError(err).Debug("Could not build redirect path")
```

Comment:
I feel like this should be an error-level message, no? This has potential to fail is outside of our control (since the response comes from GitHub), so I suppose if something goes wrong, the admin cluster would like to know that without having to turn on debug logs.

```suggestion
			logger.WithError(err).Error("Could not build redirect path")
```

Reply:
If we are bumping the level for the cluster admin, then let's try to phrase the message in a way that they would know what to do here. The error is quite mysterious otherwise.

Reply:
The only error even returned from here is just "bad param" of not passing in a valid device trust web token and nothing to do with the redirect path actually. This seems like it will be a red herring regardless.

Reply:
I think it might be better to return a default "/web" route here if the `dtw` is invalid, log that its invalid, and then continue on our way imo. that way we are always returning from this method instead of dropping out due to an error. thoughts?

Reply:
SGTM.